File: concrete/src/test/java/pcb/uwu/amount/composite/fudamental/LitresTest.java
package pcb.uwu.amount.composite.fudamental;

import org.junit.Assert;
import org.junit.Test;
import pcb.uwu.amount.composite.fundamental.Litres;
import pcb.uwu.amount.composite.fundamental.Volume;
import pcb.uwu.units.composite.fundamental.VolumeUnit;

import static org.junit.Assert.assertEquals;
import static pcb.uwu.unit.composite.fundamental.LitreUnit.LITRE;
import static pcb.uwu.unit.quantity.MeterUnit.METER;

public class LitresTest {
	@Test
	public void testConstruct() {
		VolumeUnit unit = new VolumeUnit(METER, METER, METER) {};
		Litres litres = new Litres(1);

		assertEquals("1l", litres.toString());
		assertEquals("0.001m³", litres.convertTo(unit).toString());

		Volume volume = new Volume(3.03, unit);

		Assert.assertEquals("3030.00l", volume.convertTo(LITRE).toString());
	}
}

File: implementation/src/test/java/pcb/uwu/amount/derived/geometry/LitresTest.kt
package pcb.uwu.amount.derived.geometry

import pcb.uwu.amount.base.Meters
import pcb.uwu.amount.derived.geometry.volume.CubicMeters
import pcb.uwu.amount.derived.geometry.volume.Litres
import pcb.uwu.core.Magnitude.CENTI
import pcb.uwu.core.Magnitude.DECI
import pcb.uwu.core.Magnitude.MILLI
import pcb.uwu.unit.derived.geometry.CubicMeterUnit
import pcb.uwu.unit.derived.geometry.LitreUnit
import kotlin.test.Test
import kotlin.test.assertEquals

class LitresTest
{
    @Test
    fun construct()
    {
        assertEquals(CubicMeters(0.001),
                     Litres(1) to CubicMeterUnit)
        assertEquals(Litres(3030),
                     Volume(amount = "3.03", unit = CubicMeterUnit) to LitreUnit)
    }

    @Test
    fun sphereFactory()
    {
        assertEquals(Litres("0.004188790204786390821330617958058913"),
                     Litres.FOR.sphere(Meters(10, MILLI)))
        assertEquals(Litres("4.188790204786390821330617958058913"),
                     Litres.FOR.sphere(Meters(10, CENTI)))
        assertEquals(Litres("4188.790204786390821330617958058913"),
                     Litres.FOR.sphere(Meters(10, DECI)))
    }

    @Test
    fun cubeFactory()
    {
        assertEquals(Litres(0.001),
                     Litres.FOR.cube(Meters(1, CENTI)))
        assertEquals(Litres(1),
                     Litres.FOR.cube(Meters(1, DECI)))
        assertEquals(Litres(1000),
                     Litres.FOR.cube(Meters(1)))
    }
}
